internals/cli/env.go

```diff
@@ -165,47 +158,211 @@ func (a *App) CheckStrictEnv() error {
+	if persistent {
+		cmd.PersistentFlags().DurationVarP(reference, name, shorthand, def, usage)
+	} else {
+		cmd.Flags().DurationVarP(reference, name, shorthand, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) BoolVar(reference *bool, name string, def bool, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().BoolVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().BoolVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) IntVar(reference *int, name string, def int, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().IntVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().IntVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) StringVar(reference *string, name string, def string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().StringVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().StringVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) DurationVar(reference *time.Duration, name string, def time.Duration, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().DurationVar(reference, name, def, usage)
+	} else {
+		cmd.Flags().DurationVar(reference, name, def, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) VarP(reference pflag.Value, name string, shorthand string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().VarP(reference, name, shorthand, usage)
+	} else {
+		cmd.Flags().VarP(reference, name, shorthand, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) Var(reference pflag.Value, name string, usage string, hasEnv bool, persistent bool) {
+	if persistent {
+		cmd.PersistentFlags().Var(reference, name, usage)
+	} else {
+		cmd.Flags().Var(reference, name, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+}
+
+func (cmd *CommandClause) VarPF(reference pflag.Value, name string, shorthand string, usage string, hasEnv bool, persistent bool) *pflag.Flag {
+	var flag *pflag.Flag
+	if persistent {
+		flag = cmd.PersistentFlags().VarPF(reference, name, shorthand, usage)
+	} else {
+		flag = cmd.Flags().VarPF(reference, name, shorthand, usage)
+	}
+
+	if hasEnv {
+		cmd.FlagRegister(name, usage)
+	}
+	return flag
 }
 
 // Command adds a new subcommand to this command.
-func (cmd *CommandClause) Command(name, help string) *CommandClause {
-	return &CommandClause{
-		CmdClause: cmd.CmdClause.Command(name, help),
-		name:      name,
-		app:       cmd.app,
+func (cmd *CommandClause) CreateCommand(name, help string) *CommandClause {
```

Comment:
I prefer `Command` over `CreateCommand` as it is shorter and just as clear (in my opinion). Avoiding names such as this one and `FlagRegister` seems to be another advantage of not embedding the command.

Reply:
The problem here is that it gets into a conflict with the Comand structure of cobra. At the beginning of the transition, we kept the naming to `Command`. However, we got into different errors that were only solved by renaming the function. Therefore, we changed to `CreateCommand`

Reply:
Yes, but this wouldn't be an issue if we wouldn't embed the command struct of cobra.

Reply:
If I understand correctly, would you rather have a cobra command as a field of the CommandClause, rather than as an interface it implements?

Reply:
> If I understand correctly, would you rather have a cobra command as a field of the CommandClause, rather than as an interface it implements?

Yes

Reply:
This has been done. Let us know if you have any more suggestions related to this, otherwise feel free to close the thread. :D

Reply:
Can this conversation be resolved?